src/org/parosproxy/paros/extension/report/ReportGenerator.java

```diff
@@ -121,12 +124,12 @@ public static File stringToHtml(String inxml, String infilexsl, String outfilena
 				logger.error(e.getMessage(), e);
 				// Save the xml for diagnosing the problem
 				BufferedWriter bw = null;
-	
+				JOptionPane.showMessageDialog(null, Constant.messages.getString("report.write.dialog.message"));

```

Comment:
- This should be shown only when there's view/GUI (i.e. View.isInitialised() returns true);
 - To reduce code duplication this (the view check and show of the dialogue) should be extracted to a method;
 - Now that the message is shown in several catch blocks we must return, otherwise ZAP might end up showing the warning dialogue twice (returning "File (outfilename)" at the end of the catch block is fine, it keeps the same behaviour).

There's one IOException catch block that's missing the message (the else "// No XSLT file specified, just output the XML straight to the file").